module/VuFind/src/VuFind/Tags.php

```diff
@@ -107,4 +111,14 @@ public function deleteTagsFromRecord(RecordDriver $driver, UserEntityInterface $
             $resource->deleteTag($tag, $user);
         }
     }
+
+    /**
+     * Repair duplicate tags in the database (if any).
+     *
+     * @return void
+     */
+    public function fixDuplicateTags(): void
+    {
+        $this->getDbTable('Tags')->fixDuplicateTags();
```

Comment:
The fixDuplicateTags method has a lot of business logic, so it seems like it makes sense to move it to this higher-level class instead of putting it in the low-level TagService. It's a bit complicated to refactor right now, though, so I'm just creating this wrapper as a placeholder. Things can be more effectively reorganized in #2233 after this gets merged.

Reply:
`Tags` feels like a strange class as is. Maybe it should also be renamed/refactored as a higher level `VuFind\Service\TagsService` (or `TagService`) as a part of that work?

Reply:
Yes, I think \VuFind\Tags\TagsService would be more consistent with how Ratings and Favorites are being handled. I can always retain \VuFind\Tags as a subclass and alias for backward compatibility. I'll set that up in a separate PR.